src/posix/platform/rcp_caps_diag.cpp

```diff
@@ -137,22 +141,79 @@ void RcpCapsDiag::TestSpinelCommands(Category aCategory)
     }
 }
 
+void RcpCapsDiag::ProcessDiagCommands(void)
+{
+    static const char *kDiagCommands[] = {"channel 20",
+                                          "cw start",
+                                          "cw stop",
+                                          "power 10",
+                                          "echo 112233",
+                                          "echo -n 100",
+                                          "stream start",
+                                          "stream stop",
+                                          "rawpowersetting enable",
+                                          "rawpowersetting 112233",
+                                          "rawpowersetting",
+                                          "powersettings 20",
+                                          "rawpowersetting disable"};
+
+    Output("\r\nDiag Commands :\r\n");
+
+    for (uint16_t i = 0; i < OT_ARRAY_LENGTH(kDiagCommands); i++)
+    {
+        TestDiagCommand(kDiagCommands[i]);
+    }
+}
+
+void RcpCapsDiag::TestDiagCommand(const char *aCommand)
+{
+    char    cmd[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE];
+    char    output[OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE] = {0};
+    otError error;
+
+    snprintf(cmd, sizeof(cmd), "diag %s", aCommand);
+    SuccessOrExit(error = mRadioSpinel.PlatDiagProcess(aCommand, output, sizeof(output)));
+
+    if (strstr(output, "not supported") != nullptr)
+    {
+        error = kErrorInvalidCommand;
+    }
+    else if ((strstr(output, "failed") != nullptr) && (strstr(output, "status") != nullptr))
```

Comment:
Since "status" must come after "failed", shall we change like this?
```suggestion
    else if ((strstr(output, "failed") != nullptr))
```

Reply:
In case the normal output message contains the string `failed`, I checked both the `failed` and `status` to reduce the possibility of incorrect response message parsing.